### PR TITLE
Add `param` method to `RangeQueryBuilder` and `InstantQueryBuilder`

### PR DESCRIPTION
**Pull request by alesharik**
This PR allows to add custom query parameters for  `RangeQueryBuilder` and `InstantQueryBuilder`. This is useful, for example, when interacting with VictoriaMetrics, because it supports `step` parameter in instant query.

### CODE REVIEW

**Comment by puetzp** [comment]
Fist of all thank you for the PR. 

I can certainly see the merit of having a way to add generic query parameters to a request. Your use case is one example, but I can also think of cases where query parameters or additional headers are relevant when a Prometheus (or VictoriaMetrics) server is behind a proxy that evaluates those.
The same reasoning applies to the existing method to apply additional header values per-request, see the [doc](https://docs.rs/prometheus-http-query/latest/prometheus_http_query/struct.InstantQueryBuilder.html#method.header) which is almost identical to the original implementation in [reqwest](https://docs.rs/reqwest/latest/reqwest/struct.RequestBuilder.html#method.header).

We could solve your issue by basically copying the [`query`](https://docs.rs/reqwest/latest/reqwest/struct.RequestBuilder.html#method.query) method signature in `reqwest`. 

If you agree, could you update your PR accordingly?

**Comment by alesharik** [comment]
Hi @puetzp ,
I agree, but I cannot find a way to implement such signature.
`reqwest` uses `serde_urlencoded` under the hood, which uses `Serializer` struct (not trait) from `form_urlencoded`, which supports only outputting strings. Also `serde_urlencoded` does not support already pre-made query string. So the only way to implement it will be serializing parameters to string, and then deserializing it just to pass to reqwest... which is weird.

**Comment by puetzp** [comment]
Ok I see. Then let us keep to your initial PR, but maybe change the function a little:

```rust
pub fn query(mut self, name: &'static str, value: impl ToString) -> Self {
    self.params.push((name, value.to_string()));
    self
}
```

I think it is better to let the value implement `ToString`, because this trait is automatically implemented for everything that implements `Display`. Your current function signature would not work for e.g. integer types like `u32` since `u32` does not implement `From<String>` and thus also not `Into<String>`. However `u32` and other integer types implements `Display` so `to_string()` can be called to create a string representation.
To summarize, I believe `ToString` applies to a broader range of primitives than `Into<String>`.

File: src/client.rs [hunk 1]
```diff
@@ -51,8 +51,8 @@ impl InstantQueryBuilder {
     }
 
     /// Include an additional parameter to the request.
-    pub fn param(mut self, name: &'static str, value: impl Into<String>) -> Self {
-        self.params.push((name, value.into()));
+    pub fn param(mut self, name: &'static str, value: impl ToString) -> Self {
```
Comment (puetzp) [inline src/client.rs:54]: Let's rename this to `query` in accordance with `reqwest::RequestBuilder`. Then we are good to go!

File: src/client.rs [hunk 2]
```diff
@@ -109,8 +109,8 @@ impl RangeQueryBuilder {
...
```
Comment (puetzp) [inline src/client.rs:112]: Let's rename this to `query` in accordance with `reqwest::RequestBuilder`. Then we are good to go!
